### PR TITLE
Feature/1,8,4,31,9,23,15(edited)

### PR DESCRIPTION
**Pull request by qua1ntman**
Created next components:

1. Main page with ads (task 23)
2. Header (task 31)
3. Footer (task 9)
4. Sign up page (task 6)
5. Sign in page (task 1)
6. Admin page navbar (task 15)
7. Authorization (task 8)
8. Added routes (task 34)

### CODE REVIEW

File: src/app/admin/ads-editing/ads-editing.component.css [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    font-weight: bold;
+}
+
+tr {
+    border-bottom: 1px solid rgb(153, 153, 153) ;
+}
+
+th, td {
+    padding: 5px;
+}
```
Comment (yavorovsky) [inline src/app/admin/ads-editing/ads-editing.component.css:22]: should be an extra space at the end of the file

File: src/app/admin/ads-editing/ads-editing.component.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+                <th>City</th>
+                <th>Date</th>
+                <th>Price</th>
+                <th></th>
+                <!-- <th></th> -->
+                <!-- <th></th> -->
+            </tr>
+        </thead>
+        <tbody>
+            <tr *ngFor="let ad of ads">
```
Comment (yavorovsky) [inline src/app/admin/ads-editing/ads-editing.component.html:15]: If you're gonna use server side data, you should use trackBy function in this case for rendering reasons

File: src/app/admin/ads-editing/ads-editing.component.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import { Ad } from 'types/ad';
+
+@Component({
+  selector: 'app-ads-editing',
+  templateUrl: './ads-editing.component.html',
+  styleUrls: ['./ads-editing.component.css']
+})
+export class AdsEditingComponent{
+
+  ads: Ad[] = [
```
Comment (yavorovsky) [inline src/app/admin/ads-editing/ads-editing.component.ts:11]: Move the data into a separate service and pull it from there. Please imitate an observable to get data. Once you have a configured API you just can replace it with Observable.

File: src/app/app.component.ts [hunk 4]
```diff
@@ -6,5 +6,17 @@ import { Component } from '@angular/core';
   styleUrls: ['./app.component.css']
 })
 export class AppComponent {
-  title = 'app-realt';
+  isWantToSignIn: boolean =false
```
Comment (yavorovsky) [inline src/app/app.component.ts:9]: if you set it as false it's automatically boolean. And please control spaces. It should be boolean =` `false

File: src/app/authorization/authorization.component.css [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    text-align: center;
+
+    width: 100vw;
+    height: 100vh;
+
+    z-index: 200;
+
+    background-color: rgba(145, 145, 145, .3);
+    
+}
```
Comment (yavorovsky) [inline src/app/authorization/authorization.component.css:20]: space

File: src/app/authorization/authorization.component.html [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<div (click)="changeAuth(false)" class="auth-container">
+    <app-sign-in appClickStopPropagination (changeAuth)="personLogged($event)" (changeSign)="changeSignInAndSignUp($event)" *ngIf="!isSignIn"></app-sign-in>
```
Comment (yavorovsky) [inline src/app/authorization/authorization.component.html:2]: too long string. Configure as 120 symbols limit

File: src/app/authorization/authorization.component.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  styleUrls: ['./authorization.component.css']
+})
+export class AuthorizationComponent {
+
+  isSignIn: boolean = true
+  
+  constructor() { }
+
+  @Output()
+  changeAuthHidding = new EventEmitter()
```
Comment (yavorovsky) [inline src/app/authorization/authorization.component.ts:15]: EventEmitter receives a type EventEmitter<type> ()

File: src/app/authorization/authorization.component.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+@Component({
+  selector: 'app-authorization',
+  templateUrl: './authorization.component.html',
+  styleUrls: ['./authorization.component.css']
+})
+export class AuthorizationComponent {
+
+  isSignIn: boolean = true
+  
+  constructor() { }
```
Comment (yavorovsky) [inline src/app/authorization/authorization.component.ts:12]: If you don't inject anything constructor is redundant

File: src/app/authorization/sign-in/sign-in.component.html [hunk 9]
```diff
@@ -0,0 +1,26 @@
+<div class="sign" >
+    <h1>Sign in</h1>
+    <p>or you can <span class="change-sign" (click)="change($event)">Sign up</span></p>
+    <form>
+        <div class="input-container">
+            <label for="email">Email</label>
+            <input id="email" #email="ngModel" name="email" type="text"  [(ngModel)]="loggedUser.email" required>
+        </div>
+        <br *ngIf="!email.errors?.['required'] || email.untouched">
```
Comment (yavorovsky) [inline src/app/authorization/sign-in/sign-in.component.html:9]: br self-closed tag <br/>
If you have a complicated condition move it into getter in the component. Like get hasRequiredError(): boolean { return ... }

File: src/app/authorization/sign-in/sign-in.component.html [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+        <div *ngIf="email.errors?.['required'] && email.touched" class="danger">
+            Email name is required
+        </div>
+        <div class="input-container">
+            <label for="password">Password</label>
+            <input id="password" #password="ngModel" name="password" type="password"  [(ngModel)]="loggedUser.password" required>
```
Comment (yavorovsky) [inline src/app/authorization/sign-in/sign-in.component.html:15]: input is self-closed tag <input/>

File: src/app/authorization/sign-in/sign-in.component.html [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+        </div>
+        <br *ngIf="!password.errors?.['required'] || password.untouched">
+        <div *ngIf="password.errors?.['required'] && password.touched" class="danger">
```
Comment (yavorovsky) [inline src/app/authorization/sign-in/sign-in.component.html:18]: getter

File: src/app/authorization/sign-in/sign-in.component.ts [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+})
+export class SignInComponent {
+  
+  loggedUser: UserSignInData = {
+    email: '',
+    password: ''
+  }
+
+  @Output()
+  changeSign = new EventEmitter()
```
Comment (yavorovsky) [inline src/app/authorization/sign-in/sign-in.component.ts:17]: use a generic type

File: src/app/authorization/sign-in/sign-in.component.ts [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+
+  @Output()
+  changeAuth = new EventEmitter()
+
+  constructor() { }
```
Comment (yavorovsky) [inline src/app/authorization/sign-in/sign-in.component.ts:22]: if you dont use it. remove it

File: src/app/authorization/sign-in/sign-in.component.ts [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+
+  // Change sign in on sign up form
+  change (event: any): void {
```
Comment (yavorovsky) [inline src/app/authorization/sign-in/sign-in.component.ts:25]: no any

File: src/app/authorization/sign-in/sign-in.component.ts [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+    this.changeSign.emit(true)
+    event.stopPropagation();
+
+  }
+  // "Sign in" btn work if form is valid 
+  changeAuthFunc(event: any) {
```
Comment (yavorovsky) [inline src/app/authorization/sign-in/sign-in.component.ts:31]: no any

File: src/app/authorization/sign-up/sign-up.component.css [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+
+.sign-up-button:hover {
+    background-color: #3e40bb;
+}
+
+.change-sign {
+    text-decoration: underline;
+    color: brown;
+    cursor: pointer;
+}
```
Comment (yavorovsky) [inline src/app/authorization/sign-up/sign-up.component.css:46]: space

File: src/app/authorization/sign-up/sign-up.component.html [hunk 17]
```diff
@@ -0,0 +1,59 @@
+<div class="sign" >
+    <h1>Sign up</h1>
+    <p>or you can <span class="change-sign" (click)="change($event)">Sign in</span></p>
+    <form>
+        <div class="input-container">
+            <label for="firstname" >First name</label>
+            <input id="firstname" #firstname="ngModel" name="firstName" [(ngModel)]="user.firstName" minlength="3" type="text" required>
+        </div>
+        <br *ngIf="!firstname.errors?.['minlength'] && !(firstname.errors?.['required'] && firstname.touched)">
```
Comment (yavorovsky) [inline src/app/authorization/sign-up/sign-up.component.html:9]: br/> if a condition too long, move it into component getter

File: src/app/authorization/sign-up/sign-up.component.html [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+            Last name is required
+        </div>
+        <div *ngIf="lastname?.errors?.['minlength']" class="danger">
+            <div> Minimum name length - {{lastname.errors?.['minlength']}}</div>
+        </div>
+        <div class="input-container">
+            <label for="email">Email</label>
+            <input id="email" #email="ngModel" name="email" [(ngModel)]="user.email" type="text"
+                required pattern="[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,4}$"
+            >
```
Comment (yavorovsky) [inline src/app/authorization/sign-up/sign-up.component.html:31]: it should be according to style 
`
<input

       id="email"

       #email="ngModel"
/>
`

File: src/app/authorization/sign-up/sign-up.component.ts [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+
+  user: User = {
+    firstName: '',
+    lastName: '',
+    email: '',
+    phone: '',
+    password: '', 
+  } 
+
+  constructor() { }
```
Comment (yavorovsky) [inline src/app/authorization/sign-up/sign-up.component.ts:19]: The same as mentioned before

File: src/app/client/cabinete/cabinete.component.html [hunk 20]
```diff
@@ -0,0 +1,7 @@
+<div class="cabinet-background"  (click)="closeCabinet(false)">
+    <div appClickStopPropagination class="cabinet-window">
+        <div class="cabinet-btn"><div>My ads</div></div>
+        <div class="cabinet-btn"><div>Chat with admin</div></div>
+        <div class="cabinet-btn"><div>Sign out</div></div>
+    </div>
+</div>
```
Comment (yavorovsky) [inline src/app/client/cabinete/cabinete.component.html:7]: spaces

File: src/app/client/cabinete/cabinete.component.html [hunk 21]
```diff
@@ -0,0 +1,7 @@
+<div class="cabinet-background"  (click)="closeCabinet(false)">
+    <div appClickStopPropagination class="cabinet-window">
+        <div class="cabinet-btn"><div>My ads</div></div>
+        <div class="cabinet-btn"><div>Chat with admin</div></div>
```
Comment (yavorovsky) [inline src/app/client/cabinete/cabinete.component.html:4]:
```
<div>
  <div>
```

File: src/app/client/create-new-ad-page/create-new-ad-page.component.html [hunk 22]
```diff
@@ -0,0 +1,40 @@
...
+        <div class="ad-title-input input-container">
+            <label for="title">Title</label>
+            <input id="title" type="text">
+        </div>
+        <div class="ad-describtion-input input-container">
+            <label for="describtion">Describtion</label>
+            <textarea class="description-area" name="describtion" id="describtion"></textarea>
+        </div>
+        <div class="ad-adress-input input-container">
+            <label for="adress">Adress</label>
```
Comment (yavorovsky) [inline src/app/client/create-new-ad-page/create-new-ad-page.component.html:13]: Address

File: src/app/client/create-new-ad-page/create-new-ad-page.component.html [hunk 23]
```diff
@@ -0,0 +1,40 @@
...
+            <div id="adress">
+                <select type="text">
+                    <option>1</option>
```
Comment (yavorovsky) [inline src/app/client/create-new-ad-page/create-new-ad-page.component.html:16]: Don't use default select. Take a look alternatives

File: .eslintrc.json [hunk 24]
```diff
@@ -0,0 +1,50 @@
...
+      ],
+      "parserOptions": {
+        "project": [
+          "tsconfig.json"
+        ],
+        "createDefaultProgram": true
+      },
+      "extends": [
+        "plugin:@angular-eslint/recommended",
+        "plugin:@angular-eslint/template/process-inline-templates"
```
Comment (yavorovsky) [inline .eslintrc.json:19]: add prettier plugins

File: .eslintrc.json [hunk 25]
```diff
@@ -0,0 +1,50 @@
...
+      ],
+      "rules": {
```
Comment (yavorovsky) [inline .eslintrc.json:21]: please add more rules https://javascript.plainenglish.io/how-to-configure-eslint-and-prettier-on-angular-application-87dbd767369c

File: package.json [hunk 26]
```diff
@@ -1,12 +1,23 @@
...
     "start": "ng serve",
     "build": "ng build",
     "watch": "ng build --watch --configuration development",
-    "test": "ng test"
+    "test": "ng test",
+    "lint": "ng lint"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged --relative"
```
Comment (yavorovsky) [inline package.json:14]: 👍

File: src/app/admin/admin.component.html [hunk 27]
```diff
@@ -0,0 +1,10 @@
+<div class="admin-wrapper">
+    <div class="admin-container">
+        <nav>
+            <div><a routerLink="/admin/moderation">moderation</a></div>
```
Comment (yavorovsky) [inline src/app/admin/admin.component.html:4]: Please replace routes with enums or think over about better approach

File: src/app/admin/admin.component.ts [hunk 28]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-admin',
+  templateUrl: './admin.component.html',
+  styleUrls: ['./admin.component.css']
+})
+export class AdminComponent implements OnInit {
+
+  constructor() { }
```
Comment (yavorovsky) [inline src/app/admin/admin.component.ts:10]: If you don't use ngOnInit or constructor, you should remove it

File: src/app/admin/ads-editing/ads-editing.component.html [hunk 29]
```diff
@@ -0,0 +1,26 @@
+<div class="table-moderation-container">
+    <table>
+        <thead>
+            <tr>
+                <th>Title</th>
+                <th>City</th>
+                <th>Date</th>
+                <th>Price</th>
+                <th></th>
+                <!-- <th></th> -->
```
Comment (yavorovsky) [inline src/app/admin/ads-editing/ads-editing.component.html:10]: Why has it been commented? You should leave a comment to explain it.

File: src/app/admin/appeals/appeals.component.ts [hunk 30]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-appeals',
+  templateUrl: './appeals.component.html',
+  styleUrls: ['./appeals.component.css']
+})
+export class AppealsComponent implements OnInit {
```
Comment (yavorovsky) [inline src/app/admin/appeals/appeals.component.ts:8]: If you don't use the constructor, remove it

File: src/app/app-routing.module.ts [hunk 31]
```diff
@@ -1,10 +1,56 @@
...
+
+];
 
 @NgModule({
-  imports: [RouterModule.forRoot(routes)],
+  imports: [RouterModule.forRoot(appRoutes)],
   exports: [RouterModule]
 })
 export class AppRoutingModule { }
+
```
Comment (yavorovsky) [inline src/app/app-routing.module.ts:51]: You must use only space not more

File: src/app/app.component.css [hunk 32]
```diff
@@ -0,0 +1,25 @@
+
+
```
Comment (yavorovsky) [inline src/app/app.component.css:2]: No space at the top

File: src/app/app.component.css [hunk 33]
```diff
@@ -0,0 +1,25 @@
...
+.app-wrapper {
+    display: flex;
+    flex-direction: column;
+    /* justify-content: center; */
```
Comment (yavorovsky) [inline src/app/app.component.css:6]: Why? Please add more explanation

File: src/app/app.component.css [hunk 34]
```diff
@@ -0,0 +1,25 @@
...
+    text-align: center;
+    align-items: center;
+    width: 100%;
+    min-width: 300px;
+    min-height: 84vh;
+    /* margin: 20px 0; */
+    /* margin-top: -20px; */
+    /* margin-top: 20px;
+    margin-bottom: -20px; */
+}
```
Comment (yavorovsky) [inline src/app/app.component.css:25]: A space at the bottom

File: src/app/app.component.ts [hunk 35]
```diff
@@ -6,5 +6,17 @@ import { Component } from '@angular/core';
...
+  loggedIn: boolean = false
+
+  // Show or hide authoriation
+  changeAuthoriationHidding(value: boolean): void {
+    this.isWantToSignIn=value
+  }
+
+  // Hide "Sign in" btn and chenge "Add new ad" btn in header
+  logged(value: boolean): void {
```
Comment (yavorovsky) [inline src/app/app.component.ts:18]: You should use a more meaningful name. We name functions with verbs usually

File: src/app/authorization/sign-in/sign-in.component.html [hunk 36]
```diff
@@ -0,0 +1,26 @@
...
+            Password name is required
+        </div>
+        <button class="sign-in-button" (click)="changeAuthFunc($event)">Sign in</button>
+        <p>or</p>
+        <div class="google-in another-in" (click)="changeAuthFunc($event)">Join with Google</div>
+        <div class="github-in another-in" (click)="changeAuthFunc($event)">Join with Github</div>
+    </form>
+</div>
```
Comment (yavorovsky) [inline src/app/authorization/sign-in/sign-in.component.html:26]: space

File: src/app/authorization/sign-up/sign-up.component.ts [hunk 37]
```diff
@@ -0,0 +1,52 @@
...
+    if (this.isFormFilled()) {
+      event.stopPropagation();
+      this.changeAuth.emit(false)
+    }
+  }
+
+  // Form validation for "Sign up" btn
+  isFormFilled(): boolean {
+    return Boolean(
+      this.user.firstName.length >= 3 &&
```
Comment (yavorovsky) [inline src/app/authorization/sign-up/sign-up.component.ts:44]: Please use validators separately. Too much logic for a function

File: src/app/authorization/sign-up/sign-up.component.ts [hunk 38]
```diff
@@ -0,0 +1,52 @@
...
+  constructor() { }
+
+  @Output()
+  changeSign = new EventEmitter()
+
+  @Output()
+  changeAuth = new EventEmitter()
+
+  // Change sign up on sign in form
+  change (event: any): void {
```
Comment (yavorovsky) [inline src/app/authorization/sign-up/sign-up.component.ts:28]: no any if it's possible

File: src/app/authorization/sign-up/sign-up.component.ts [hunk 39]
```diff
@@ -0,0 +1,52 @@
...
+    password: '', 
+  } 
+
+  constructor() { }
+
+  @Output()
+  changeSign = new EventEmitter()
+
+  @Output()
+  changeAuth = new EventEmitter()
```
Comment (yavorovsky) [inline src/app/authorization/sign-up/sign-up.component.ts:25]: generic

**Review by yavorovsky — CHANGES_REQUESTED** [review]

File: src/app/client/create-new-ad-page/create-new-ad-page.component.html [hunk 40]
```diff
@@ -0,0 +1,40 @@
...
+                <input type="text" placeholder="Street">
+            </div>
+
+        </div>
+        <div class="ad-photo">
+            <button class="add-photo-btn">Add photo</button>
+        </div>
+        <button class="create-ad-button">Create</button>
+    </form>
+</div>
```
Comment (yavorovsky) [inline src/app/client/create-new-ad-page/create-new-ad-page.component.html:40]: space

File: src/app/client/create-new-ad-page/create-new-ad-page.component.ts [hunk 41]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-create-new-ad-page',
+  templateUrl: './create-new-ad-page.component.html',
+  styleUrls: ['./create-new-ad-page.component.css']
+})
+export class CreateNewAdPageComponent implements OnInit {
+
+  constructor() { }
```
Comment (yavorovsky) [inline src/app/client/create-new-ad-page/create-new-ad-page.component.ts:10]: !

File: src/app/client/edit-ad-page/edit-ad-page.component.ts [hunk 42]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-edit-ad-page',
+  templateUrl: './edit-ad-page.component.html',
+  styleUrls: ['./edit-ad-page.component.css']
+})
+export class EditAdPageComponent implements OnInit {
+
+  constructor() { }
```
Comment (yavorovsky) [inline src/app/client/edit-ad-page/edit-ad-page.component.ts:10]: !

File: src/app/directives/stop-propagination/click-stop-propagination.directive.ts [hunk 43]
```diff
@@ -0,0 +1,14 @@
+import { Directive, HostListener} from '@angular/core';
+
+@Directive({
+  selector: '[appClickStopPropagination]'
+})
+export class ClickStopPropaginationDirective {
+
+  @HostListener("click", ["$event"])
+  public onClick(event: any): void
```
Comment (yavorovsky) [inline src/app/directives/stop-propagination/click-stop-propagination.directive.ts:9]: no any

Comment (yavorovsky) [reply src/app/directives/stop-propagination/click-stop-propagination.directive.ts:9]: actually a cool directive!

File: src/app/footer/footer.component.css [hunk 44]
```diff
@@ -0,0 +1,75 @@
...
+    text-align: center;
+
+    width: 1440px;
+}
+
+.contact-container {
+    display: flex;
+    flex-direction: column;
+    justify-content: space-around;
+    height: 60px;
```
Comment (yavorovsky) [inline src/app/footer/footer.component.css:26]: try not to use static sizes

File: src/app/footer/footer.component.css [hunk 45]
```diff
@@ -0,0 +1,75 @@
...
+
+.telegram {
+    background: url(../../app/pngs/telegram.png) no-repeat center;
+    background-size: 100%;
+}
+
+.vk {
+    background: url(../../app/pngs/vk.png) no-repeat center;
+    background-size: 100%;
+}
```
Comment (yavorovsky) [inline src/app/footer/footer.component.css:75]: space

File: src/app/footer/footer.component.html [hunk 46]
```diff
@@ -0,0 +1,17 @@
...
+            <div class="footer-text">Our socials</div>
+            <div class="social-container">
+                <a href="http://www.facebook.com" target="_blank" class="social facebook"></a>
+                <a href="http://www.instagram.com" target="_blank" class="social instagram" ></a>
+                <a href="http://www.telegram.org" target="_blank" class="social telegram" ></a>
+                <a href="http://www.vk.com" target="_blank" class="social vk" ></a>
+            </div>
+        </div>
+    </div> 
+</footer>
```
Comment (yavorovsky) [inline src/app/footer/footer.component.html:17]: space

File: src/app/footer/footer.component.ts [hunk 47]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-footer',
+  templateUrl: './footer.component.html',
+  styleUrls: ['./footer.component.css']
+})
+export class FooterComponent implements OnInit {
+
+  constructor() { }
```
Comment (yavorovsky) [inline src/app/footer/footer.component.ts:10]: !

File: src/app/header/header.component.css [hunk 48]
```diff
@@ -0,0 +1,79 @@
+.header-container {
+    display: flex;
+
```
Comment (yavorovsky) [inline src/app/header/header.component.css:3]: why space ?

File: src/app/header/header.component.css [hunk 49]
```diff
@@ -0,0 +1,79 @@
...
+    justify-content: center;
+
+    width: 100%;
+    min-height: 7vh;
+
```
Comment (yavorovsky) [inline src/app/header/header.component.css:8]: why space ?

File: src/app/header/header.component.css [hunk 50]
```diff
@@ -0,0 +1,79 @@
...
+    background-color: #2E5DFA;
+}
+
+.header {
+    display: flex;
+    justify-content: space-around;
+
+    width: 1440px;
```
Comment (yavorovsky) [inline src/app/header/header.component.css:16]: please don't use static sizes

File: src/app/header/header.component.css [hunk 51]
```diff
@@ -0,0 +1,79 @@
...
+    padding: 15px 0 ;
+    border-bottom: 2px solid #162D7A;
+}
+
+.logo-container {
+    height: 50px;
```
Comment (yavorovsky) [inline src/app/header/header.component.css:22]: same

File: src/app/header/header.component.css [hunk 52]
```diff
@@ -0,0 +1,79 @@
...
+
+.logo {
+    height: 50px;
+}
+
+.header-buttons {
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    width: 180px;
```
Comment (yavorovsky) [inline src/app/header/header.component.css:34]: same

File: src/app/header/header.component.css [hunk 53]
```diff
@@ -0,0 +1,79 @@
...
+    height: 40px;
```
Comment (yavorovsky) [inline src/app/header/header.component.css:35]: same

File: src/app/header/header.component.css [hunk 54]
```diff
@@ -0,0 +1,79 @@
...
+    border: 2px solid rgb(2, 0, 114);
+    border-radius: 50%;
+    background-color: rgb(89, 29, 255);
+    cursor: pointer;
+
+}
+
+.cabinet-button:hover {
+    background-color: rgb(129, 85, 252);
+}
```
Comment (yavorovsky) [inline src/app/header/header.component.css:79]: space

File: src/app/header/header.component.html [hunk 55]
```diff
@@ -0,0 +1,15 @@
+<header class="header-container">
+    <div class="header">
+        <!-- Just random logo -->
+        <div class="logo-container" routerLink="/ads">
+            <img src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQ0kJ1egqdu7Z3-HEJZxg6rdtotw3_qktoYzg&usqp=CAU" class="logo">
```
Comment (yavorovsky) [inline src/app/header/header.component.html:5]: use a logo URL in readonly property of the Component file. Much better to move it out the file into a more appropriate file as "constants" or something like that.

File: src/app/header/header.component.html [hunk 56]
```diff
@@ -0,0 +1,15 @@
...
+        </div>
+        <div class="header-buttons">
+            <button *ngIf="loggedIn" class="add-new-button" routerLink="create">+ Add new ad</button>
```
Comment (yavorovsky) [inline src/app/header/header.component.html:8]:
```
<ng-container ngIf="loggedIn">
.....
</ng-container>

<ng-container ngIf="!loggedIn">
.....
</ng-container>
```

**Review by yavorovsky — CHANGES_REQUESTED** [review]

File: src/app/header/header.component.ts [hunk 57]
```diff
@@ -0,0 +1,31 @@
+import { Component, Output, EventEmitter, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-header',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.css']
+})
+export class HeaderComponent {
+
+  openCabinet: boolean = false
```
Comment (yavorovsky) [inline src/app/header/header.component.ts:10]: openCabinet = false. The type will be set automatically

File: src/app/header/header.component.ts [hunk 58]
```diff
@@ -0,0 +1,31 @@
...
+
+  @Output()
+  changeAuthHidding = new EventEmitter()
```
Comment (yavorovsky) [inline src/app/header/header.component.ts:13]: generic

File: src/app/header/header.component.ts [hunk 59]
```diff
@@ -0,0 +1,31 @@
...
+
+  constructor() { }
+
+  @Input()
+  loggedIn!: boolean;
+
+  // Hide "Sign in" btn and chenge "Add new ad" btn 
+  changeAuth(): void {
+    this.changeAuthHidding.emit(true)
```
Comment (yavorovsky) [inline src/app/header/header.component.ts:22]: you can use it in template

File: src/app/header/header.component.html [hunk 60]
```diff
@@ -0,0 +1,15 @@
...
+            <button *ngIf="!loggedIn" class="add-new-button" (click)="changeAuth()">+ Add new ad</button>
+            <button *ngIf="!loggedIn" class="sign-in-button" (click)="changeAuth()">Sign in</button>
```
Comment (yavorovsky) [inline src/app/header/header.component.html:10]: (click) ="changeAuthHidding.emit(true)"

File: src/app/header/header.component.html [hunk 61]
```diff
@@ -0,0 +1,15 @@
...
+            <div *ngIf="loggedIn" class="cabinet-button" (click)="openCabinetFunc(true)">M</div>
```
Comment (yavorovsky) [inline src/app/header/header.component.html:11]: the same

File: src/app/header/header.component.ts [hunk 62]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  // Open menu (as cabinet)
+  openCabinetFunc(value: boolean): void {
```
Comment (yavorovsky) [inline src/app/header/header.component.ts:26]: this function is redundant. You may use it in template. Please review all cases that might exist as well

File: src/app/main/main.component.css [hunk 63]
```diff
@@ -0,0 +1,97 @@
...
+.ad {
+    position: relative;
+    align-self: center;
+    justify-self: center;
+
+    display: flex;
+    flex-direction: column;
+    justify-content: space-around;
+    align-items: center;
+    width: 300px;
```
Comment (yavorovsky) [inline src/app/main/main.component.css:19]: please avoid static sizes

File: src/app/main/main.component.css [hunk 64]
```diff
@@ -0,0 +1,97 @@
...
+    text-align: center;
+    transition: all .2s;
+
+    cursor: pointer;
+}
+
+.minimap:hover {
+    background-color: rgb(4, 185, 185);
+
+}
```
Comment (yavorovsky) [inline src/app/main/main.component.css:97]: space

File: src/app/main/main.component.html [hunk 65]
```diff
@@ -0,0 +1,14 @@
...
+        <div class="ad-information">
+            <div class="ad-title-div">{{ad.title}}</div>
+            <div class="ad-cost-div">{{ad.price}}$</div>
+            <div class="ad-location-div">{{ad.location}}</div>
+        </div>
+    </div>
+    <div routerLink="/map" class="minimap">
+        Watch on map
+    </div>
+</div>
```
Comment (yavorovsky) [inline src/app/main/main.component.html:14]: space

File: src/app/main/main.component.html [hunk 66]
```diff
@@ -0,0 +1,14 @@
...
+    <div *ngFor="let ad of ads" class="ad">
+        <div class="ad-picture" [style.background-color]="ad.pic" ></div>
+        <div class="ad-date">{{ad.date}}</div>
+        <div class="ad-information">
+            <div class="ad-title-div">{{ad.title}}</div>
+            <div class="ad-cost-div">{{ad.price}}$</div>
+            <div class="ad-location-div">{{ad.location}}</div>
+        </div>
+    </div>
+    <div routerLink="/map" class="minimap">
```
Comment (yavorovsky) [inline src/app/main/main.component.html:11]: enum

File: src/app/main/main.component.ts [hunk 67]
```diff
@@ -0,0 +1,71 @@
...
+
+@Component({
+  selector: 'app-main',
+  templateUrl: './main.component.html',
+  styleUrls: ['./main.component.css']
+})
+export class MainComponent{
+
+  //data sample
+  ads: Ad[]=[
```
Comment (yavorovsky) [inline src/app/main/main.component.ts:12]: move all data into separate service

Comment (yavorovsky) [reply src/app/main/main.component.ts:12]: try to follow this patterns https://medium.com/@ankit.sinhal/mvc-mvp-and-mvvm-design-pattern-6e169567bbad

File: src/app/main/main.component.ts [hunk 68]
```diff
@@ -0,0 +1,71 @@
...
+      title: 'penthouse',
+      price: 445000,
+      location: 'Minsk'
+    },
+  ]
+
+  constructor() { }
+
+  randomColor(): string {
+    return `rgb(${Math.round(Math.random()*255)}, ${Math.round(Math.random()*255)}, ${Math.round(Math.random()*255)})`
```
Comment (yavorovsky) [inline src/app/main/main.component.ts:67]: it should be a util function as it's not a part of the component (logically)

Comment (yavorovsky) [reply src/app/main/main.component.ts:67]: Move it out of the file into a ".utils.ts" file

File: src/app/map-page/map-page.component.ts [hunk 69]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-map-page',
+  templateUrl: './map-page.component.html',
+  styleUrls: ['./map-page.component.css']
+})
+export class MapPageComponent implements OnInit {
+
+  constructor() { }
```
Comment (yavorovsky) [inline src/app/map-page/map-page.component.ts:10]: !

File: types/ad.ts [hunk 70]
```diff
@@ -0,0 +1,7 @@
+export interface Ad {
+    pic: string,
+    date: string,
+    title: string,
+    price: number,
+    location: string
+}
```
Comment (yavorovsky) [inline types/ad.ts:7]: space

File: types/user-sign-in-data.ts [hunk 71]
```diff
@@ -0,0 +1,4 @@
+export interface UserSignInData {
+    email: string
+    password: string
+}
```
Comment (yavorovsky) [inline types/user-sign-in-data.ts:4]: space

File: types/user-sign-up-data.ts [hunk 72]
```diff
@@ -0,0 +1,7 @@
+export interface User {
+    firstName: string,
+    lastName: string,
+    email: string,
+    phone: string,
+    password: string,
+}
```
Comment (yavorovsky) [inline types/user-sign-up-data.ts:7]: space

File: types/user.ts [hunk 73]
```diff
@@ -0,0 +1,7 @@
+export interface User {
+    firstName: string,
+    lastName: string,
+    email: string,
+    phone: string,
+    password: string,
+}
```
Comment (yavorovsky) [inline types/user.ts:7]: space

**Review by yavorovsky — CHANGES_REQUESTED** [review]

File: src/app/main/main.component.css (hunk 63 above)
Comment (qua1ntman) [reply src/app/main/main.component.css:19]: I feel 300px for ad is optimized size cause there are not many devices with width smaller than 300-350px

File: src/app/header/header.component.css (hunk 52 above)
Comment (qua1ntman) [reply src/app/header/header.component.css:34]: Here i will change sizes only if i will add media queries

File: src/app/header/header.component.css (hunk 53 above)
Comment (qua1ntman) [reply src/app/header/header.component.css:35]: It is a good size for buttons container

File: src/app/authorization/sign-in/sign-in.component.html (hunk 9 above)
Comment (qua1ntman) [reply src/app/authorization/sign-in/sign-in.component.html:9]: According to habr br, input, img tags must be opened

File: src/app/authorization/sign-up/sign-up.component.html (hunk 17 above)
Comment (qua1ntman) [reply src/app/authorization/sign-up/sign-up.component.html:9]: As I khow, 'br />' is valid only in xml. In html you cannot close such tags as br, input, img

File: src/app/authorization/sign-up/sign-up.component.html (hunk 18 above)
Comment (qua1ntman) [reply src/app/authorization/sign-up/sign-up.component.html:31]: As I khow, 'br />' is valid only in xml. In html you cannot close such tags as br, input, img

**Pull request merged**